packages/react/src/types.ts

```diff
@@ -71,7 +71,11 @@ type ReactNode =
   | null
   | undefined
 
-export type ShorthandValue<P> = ReactNode | Props<P>
+export type ShorthandRenderProp<P> = (Component: React.ElementType, props: P) => React.ReactNode
+
+export type ShorthandValue<P extends Props> =
+  | ReactNode
+  | Props<P> & { children?: P['children'] | ShorthandRenderProp<P> }
```

Comment:
@miroslavstastny I am not sure about this typing, any thoughts?

Reply:
I like it 😃